src/averageAges.js

```diff
@@ -14,12 +15,19 @@
  *
  * @return {number}
  */
+
+const reducer = (prev, curnt) => prev + curnt;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = people.filter(({ sex }) => sex === 'm');
+
+  century && (men = men
+    .filter(({ died }) => Math.ceil(died / 100) === century));
+
+  const lifetimeArr = men.map(({ born, died }) => died - born);
+  const sumAges = lifetimeArr.reduce(reducer);
```

Comment:
It is not required, but i just want to suggest you try to solve this function in one iteration.
In most cases when we should filter and than map same array it is possible to replace on reduce method and do everything in one iteration.
Hint: accamulator can be any structure of data.